platforms/aws/giantnetes/main.tf

```diff
@@ -0,0 +1,91 @@
+module "container_linux" {
+  source = "../../../modules/container-linux"
+
+  coreos_channel = "${var.container_linux_channel}"
+  coreos_version = "${var.container_linux_version}"
+}
+
+# Get ami ID for specific Container Linux version.
+data "aws_ami" "coreos_ami" {
+  filter {
+    name   = "name"
+    values = ["CoreOS-${var.container_linux_channel}-${module.container_linux.coreos_version}-*"]
+  }
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name   = "owner-id"
+    values = ["595879546273"]
+  }
+}
+
+module "dns" {
+  source = "../../../modules/aws/dns"
+
+  cluster_name     = "${var.cluster_name}"
+  root_dns_zone_id = "${var.root_dns_zone_id}"
+  zone_name        = "${var.base_domain}"
+}
+
+module "vpc" {
+  source = "../../../modules/aws/vpc"
+
+  cluster_name     = "${var.cluster_name}"
+  subnet_bastion_0 = "${var.subnet_bastion_0}"
+  subnet_bastion_1 = "${var.subnet_bastion_1}"
+  subnet_elb_0     = "${var.subnet_elb_0}"
+  subnet_elb_1     = "${var.subnet_elb_1}"
+  subnet_worker_0  = "${var.subnet_worker_0}"
+  subnet_worker_1  = "${var.subnet_worker_1}"
+  subnet_vault_0   = "${var.subnet_vault_0}"
+  vpc_cidr         = "${var.vpc_cidr}"
+}
+
+# Create S3 bucket for ignition configs.
+module "s3" {
+  source = "../../../modules/aws/s3"
+
+  aws_account  = "${var.aws_account}"
+  cluster_name = "${var.cluster_name}"
+}
+
+locals {
+  ignition_users = "${file("${path.module}/../../../ignition/users.yaml")}"
+}
+
+# Generate ignition config for bastions.
+data "template_file" "bastion" {
+  template = "${file("${path.module}/../../../ignition/bastion.yaml.tmpl")}"
+}
+
+# Convert ignition config to raw json.
+data "ct_config" "bastion" {
```

Comment:
here is the interesting part. I've looked on possibility to use [ignition_config](https://www.terraform.io/docs/providers/ignition/d/config.html), but it does not fit well for our case. Problem is that i need to define every particular part of ignition template in terraform. e.g. how to define [systemd unit](https://www.terraform.io/docs/providers/ignition/d/systemd_unit.html#example-usage). I don't like it.

So i took alternative approach.
1) generate ignition from yaml-based "container linux configuration" files (similar to cloud-config)
2) convert to raw json ignition with `ct` provider. (`ct` is official coreos tool to convert from yaml to ignition)

One important thing: [ct terraform provider](https://github.com/coreos/terraform-provider-ct/issues/21) is still not in official terraform repos. Latest release 0.2.0 had issues with user "groups", but latest dev version worked fine for me.